sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2045,6 +2047,21 @@ object EliminateUnions extends Rule[LogicalPlan] {
 }
 
 /**
+ * Converts foldable numeric expressions to integers in [[GlobalLimit]] and [[LocalLimit]] operators
+ */
+object ResolveLimits extends Rule[LogicalPlan] {
```

Comment:
Thanks for trying this out! Yea looks like it's doable. Let's remove it first and do it in follow-ups, to make this PR surgical.